### PR TITLE
make cache config available for OSS version

### PR DESCRIPTION
**Pull request by sansolovyov**
I wrote cache configuration API like OSS version doesn't need it, but it does - there should be an ability to configure `ttl` strategy, only for the root.

Two questions I'm not sure about:
- if `/api/caching/` is a good location, maybe `/api/setting/caching/` is better
- dances with schemas make them less static and because of that OpenAPI generates only OSS-version docs; would be nice to combine `defenterprise` + `defendpoint` somehow, but both of their implementations are not very composable, so it needs thought.

Part of #36847

### CODE REVIEW

File: src/metabase/api/caching.clj [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  [schema]
+  (walk/prewalk
+   (fn [x]
+     (if (and (vector? x) (= (first x) :map))
+       (into [] (remove #(:internal (meta %))) x)
+       x))
+   schema))
+
+;; TODO: declaring schema as defenterprise will make OpenAPI docs invalid for an EE version
+(defenterprise CacheStrategy
```
Comment (crisptrutski) [inline src/metabase/api/caching.clj:43]: I'm confused by this comment.. have we decided it's OK to break the OpenAPI docs for EE for now?

File: src/metabase/api/caching.clj [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  "Schema for a caching strategy for the API"
+  []
+  (drop-internal-fields (CacheStrategy)))
+
+(def ^:private CachingModel [:enum "root" "database" "dashboard" "question"])
+
+(defn- assert-valid-models [model ids premium?]
+  (cond
+    (= model "root")
+    (when-not (= ids [0])
```
Comment (crisptrutski) [inline src/metabase/api/caching.clj:68]: Based on the last PR's discussion, I think it's worth changing to `(some zero? ids)` 😄

**Comment by crisptrutski** [comment]
Makes sense, just want to understand what's going on with the TODO before approving.

File: src/metabase/api/caching.clj (hunk 2 above)
Comment (piranha) [reply src/metabase/api/caching.clj]: done

File: src/metabase/api/caching.clj (hunk 1 above)
Comment (piranha) [reply src/metabase/api/caching.clj]: do you like the new comment better? :)

Comment (crisptrutski) [reply src/metabase/api/caching.clj]: much clearer, thanks!

**Review by crisptrutski — APPROVED** [review]

**Pull request merged**